src/main/java/uk/gov/hmcts/ccd/domain/service/accessprofile/filter/FilterRoleAssignmentsImpl.java

```diff
@@ -0,0 +1,139 @@
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignments;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+import uk.gov.hmcts.ccd.domain.model.std.CaseDataContent;
+import uk.gov.hmcts.ccd.domain.service.AccessControl;
+import uk.gov.hmcts.ccd.domain.service.common.SecurityClassificationUtils;
+
+import static uk.gov.hmcts.ccd.domain.service.common.SecurityClassificationUtils.caseHasClassificationEqualOrLowerThan;
+
+@Slf4j
+@Component
+public class FilterRoleAssignmentsImpl implements FilterRoleAssignments, AccessControl {
+
+    @Override
+    public RoleAssignments filter(RoleAssignments roleAssignments, CaseDetails caseDetails) {
+        log.info("Filter role assignments for case {}", caseDetails.getReference());
+
+        List<RoleAssignment> roleAssignmentList = roleAssignments.getRoleAssignments().stream()
+            .filter(roleAssignment -> validateStartAndEndDate(roleAssignment))
+            .filter(roleAssignment -> validateSecurityValidation(caseDetails, roleAssignment))
+            .filter(roleAssignment -> validateCaseId(caseDetails, roleAssignment))
+            .filter(roleAssignment -> validateJurisdiction(caseDetails, roleAssignment))
+            .filter(roleAssignment -> validateLocation(roleAssignment))
+            .filter(roleAssignment -> validateRegion(roleAssignment))
+            .collect(Collectors.toList());
+
+        return RoleAssignments.builder().roleAssignments(roleAssignmentList).build();
+    }
+
+    @Override
+    public RoleAssignments filter(RoleAssignments roleAssignments, CaseDataContent caseDataContent) {
+        log.info("Filter role assignments for case event {}", caseDataContent.getEvent().getEventId());
+
+        List<RoleAssignment> roleAssignmentList = roleAssignments.getRoleAssignments().stream()
+            .filter(roleAssignment -> validateStartAndEndDate(roleAssignment))
+            .collect(Collectors.toList());
+        return RoleAssignments.builder().roleAssignments(roleAssignmentList).build();
+    }
+
+    @Override
+    public RoleAssignments filter(RoleAssignments roleAssignments, CaseTypeDefinition caseTypeDefinition) {
+        log.info("Filter role assignments for case type {}", caseTypeDefinition.getName());
+
+        List<RoleAssignment> roleAssignmentList = roleAssignments.getRoleAssignments().stream()
+            .filter(roleAssignment -> validateStartAndEndDate(roleAssignment))
+            .collect(Collectors.toList());
+        return RoleAssignments.builder().roleAssignments(roleAssignmentList).build();
+    }
+
+    private boolean validateJurisdiction(CaseDetails caseDetails, RoleAssignment roleAssignment) {
+        log.debug("Apply filter on jurisdiction {} for role assignment {}",
+            caseDetails.getJurisdiction(),
+            roleAssignment.getId());
+        roleAssignment.getMatchingResults().setValidJurisdiction(
```

Comment:
I wouldn't add the matching results inside the RoleAssignment. Don't see the reason. I would have them as a separate entities returned by the filtering service.
i.e. the `filter` methods return a `RoleAssignmentFilteringResult` object, which contains the new list of RoleAssigment, and a list of `RoleMatchingResult`s.
One benefit is that in the future if needed we can add additional properties on this `RoleAssignmentFilteringResult`